app/src/main/java/ch/epfl/sweng/erpa/services/dummy/database/DummyDataService.java

```diff
@@ -0,0 +1,111 @@
+package ch.epfl.sweng.erpa.services.dummy.database;
+
+import android.content.Context;
+import android.util.Log;
+
+import com.annimon.stream.Collectors;
+import com.annimon.stream.Optional;
+import com.annimon.stream.Stream;
+import com.annimon.stream.function.Function;
+
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.OutputStreamWriter;
+import java.util.Arrays;
+import java.util.Set;
+
+import ch.epfl.sweng.erpa.model.UidObject;
+import ch.epfl.sweng.erpa.services.DataService;
+
+import static android.content.ContentValues.TAG;
+
+public abstract class DummyDataService<T extends UidObject> implements DataService<T>
+{
+    private final static String SAVED_DATA_FILE_EXTENSION = ".yaml";
+    private final Function<File,T> fileFetcher;
+
+    private final File dataDir;
+    private final Class<T> tClass;
+
+    DummyDataService(Context ctx, Class<T> tClass) {
+        this.tClass=tClass;
+        this.dataDir = new File(ctx.getFilesDir(),dataFolder());
+        this.fileFetcher = file -> fetchExistingDataFromFile(file,tClass);
+        if(!dataDir.mkdir())
+        {
+            if(!dataDir.isDirectory())
+            {
+                throw new IllegalStateException(tClass.getName() + " data folder (\"" + dataDir.getAbsolutePath() + "\") exists and is not a folder!");
+            }
+        }
+    }
+
+    abstract String dataFolder();
+
+    @Override
+    public Optional<T> getOne(String gid) {
+        File gameFile = new File(dataDir, gid + SAVED_DATA_FILE_EXTENSION);
+        if (gameFile.isDirectory()) {
+            throw new IllegalStateException("Folder " + gameFile.getAbsolutePath() + " exists!");
+        } else {
+            return Optional.of(gameFile)
+                    .filter(File::exists)
+                    .filter(File::isFile)
+                    .map(fileFetcher);
+        }
+    }
+
+
+    @Override
+    public void saveOne(T t) {
+        String gid = t.getUid();
+        try {
+            File gameFile = new File(dataDir, gid + SAVED_DATA_FILE_EXTENSION);
+            if (!gameFile.exists()) {
+                //noinspection ResultOfMethodCallIgnored
+                gameFile.createNewFile();
```

Comment:
What about
```
Optional.of(new File(....))
    .filter(File::exists)
    .map(fp -> new OutputStreamWriter(new FileOutputStream ...))
    .map(writer -> new Yaml().dump(t, writer)
    .getOrElseThrow(() -> new IllegalStateException("Couldn't save file"))
```

Reply:
Doesn't cover the case where the file doesn't exist but can be created